### ISSUE TITLE
[Bug] Ruby Parsing issue

### ISSUE THREAD

**Issue by ankit-privado** [joernio/joern#3043]
Describe the bug

Ruby parser fails for the below code
Please keep the conversation English! Make it easy for us and share your input code or a minified example.

To Reproduce
Load [this](https://github.com/mfssaura/HospitalManagementProject/blob/684696e569e1935fcff0d6be8f3ce3df83bbdacb/app/models/user.rb#L7) code in the Ruby parser and see it failing.

Expected behavior
Parsing should pass

Screenshots
NA

Desktop (please complete the following information):
NA

Additional context
NA

**Comment by xavierpinho** [comment]
The issue occurs on line 53, with `->`.

**Comment by ankit-privado** [comment]
@xavierpinho as you mentioned issue occurs on line 53 with `->`, this issue is still there. should i create new ticket for this? 

<img width="1455" alt="Screenshot 2023-07-18 at 4 59 31 PM" src="https://github.com/joernio/joern/assets/118803988/723f08c6-70ce-4d18-a3bb-d2fa0c0819fa">

**Comment by xavierpinho** [comment]
@ankit-privado yes, please try to reduce the sample so it's manageable. For instance, with this sample I detected we are not properly parsing `class << self`, but there are other issues as well.

**Comment by ankit-privado** [comment]
@xavierpinho yes sure.

**Comment by xavierpinho** [comment]
@ankit-privado Just checked now and with #3131 added this sample parses without issues. Let me know if I overlooked something.

**Comment by ankit-privado** [comment]
Yes right issue is resolved now, thank you @xavierpinho for quick fix.
